### DISCUSSION TITLE
[BUG] Get-PnPTenantServicePrincipal fails with "Unkown Error"

### DISCUSSION THREAD

**Discussion by EinmalIM** [pnp/powershell#3879] [Q&A]
I use Connect-PnPOnline to connect to https://mytenant-admin.sharepoint.com with -interactive and use a global admin to login.

I call Get-PnPTenantServicePrincipal and get the error "Unkown error"

With PnP debug log I get this output:

pwsh Error: 0 : 2024-04-09 16:34:46.8808        [PnP.Framework] [0]     [Error] ExecuteQuery threw following exception: Microsoft.SharePoint.Client.ServerException: Unknown Error
   at Microsoft.SharePoint.Client.ClientRequest.ProcessResponseStream(Stream responseStream)
   at Microsoft.SharePoint.Client.ClientRequest.ProcessResponse()
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryToServerAsync(ChunkStringBuilder sb)
   at Microsoft.SharePoint.Client.ClientRequest.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientRuntimeContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContext.ExecuteQueryAsync()
   at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryImplementation(ClientRuntimeContext clientContext, Int32 retryCount, String userAgent)
ServerErrorCode: -1
ServerErrorTypeName: Microsoft.SharePoint.Client.UnknownError
ServerErrorTraceCorrelationId: 54571da1-6033-8000-8b0c-ac72e731d398
ServerErrorValue:
ServerErrorDetails:
.       0ms
Get-PnPTenantServicePrincipal: Unknown Error

### Expected behavior 
I expect the call to succeed.

### What is the version of the Cmdlet module you are running?
I tried that with PnP PowerShell 2.2.0 and 2.4.0

### Which operating system/environment are you running PnP PowerShell on?
- [x ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

I have three SharePoint tenanst where the command fails this way and one where the command is succeding.

The App Reg "SharePoint Online Client Extensibility Web Application Principal" is existing in all of them.

What can cause that error?

**Comment by fredjj63** [comment]
We have the same issue, other commands like Get-PnPTenantServicePrincipalPermissionGrants fail with "Unkown error" as well.
Approve-PnPTenantServicePrincipalPermissionRequest throws following error: "The service principal for permssion request could not be found." 
"SharePoint Online Client Extensibility Web Application Principal" exists on our tenant as well and approving API access requests manually through admin center works fine.

**Comment by ryder-cayden** [comment]
We are also getting the same error for the following cmdlets:
- Get-PnPTenantServicePrincipal
- Enable-PnPTenantServicePrincipal
- Disable-PnPTenantServicePrincipal
